lib/tasks/parse_gtfs.js

```diff
@@ -0,0 +1,154 @@
+// Here lies my janky CSV parser for Google Transit Feed Spec files.
+// It parses out scheduled times, intended to help the near-end stations.
+// This initial pass relies on:
+// - the gtfs files existing in a 'google_transit' folder in app root
+//   - unzip them after downloading from marta's website
+// - route IDs currently starting with '8' for trains
+// - various hard-coded indices in the CSV files (watch these change)
+
+var fs = require('fs');
+var _ = require('lodash');
+var promise = require('q');
+
+var readFile = promise.nfbind(fs.readFile);
+
+// of course they couldn't use the same station names
+// as in the realtime API
+var station_name_mapping = {
+  'LAKEWOOD-FT MCPHERSON  STATION': 'LAKEWOOD STATION',
+  'EDGEWOOD-CANDLER PARK STATION': 'EDGEWOOD STATION',
+  'INMAN PARK-REYNOLDSTOWN STATION': 'INMAN PARK STATION',
+  'DOME-GWCC-PHILIPS ARENA-CNN STATION': 'OMNI DOME STATION',
+  'BROOKHAVEN-OGLETHORPE STATION': 'BROOKHAVEN STATION'
+};
+var stationNameFor = function(name) {
+  return (station_name_mapping[name] || name).toLowerCase();
+};
+
+var dow = ['mon', 'tue', 'wed', 'thu', 'fri', 'sat', 'sun'];
+var getDates = readFile('./google_transit/calendar.txt', {encoding: 'utf8'}).then(function(data) {
+  var ret = {};
+  data.split('\r\n').forEach(function(c) {
+    c = c.split(',');
+    for (var i = 0; i < dow.length; i++) {
+      if (c[i+1] === '1') {
+        ret[dow[i]] = c[0];
+      }
+    }
+  });
+  return ret;
+});
+
+var findTrips = readFile('./google_transit/trips.txt', {encoding: 'utf8'}).then(function(trips) {
+  var filtered_trips = _.filter(trips.split("\r\n"), function(r) {
+    return r.indexOf('8') === 0; 
+  });
+  var res = {};
+  filtered_trips.forEach(function(t) {
+    t = t.split(',');
+    var desc = t[3].split(' ');
+    res[t[2]] = {id: t[2], dir: desc[1][0], line: desc[0], service: t[1], stops: []};
+  });
+  return res;
+});
+
+var stopTimes = findTrips.then(function(trips) {
+  var resolve;
+  // bluebird compatible, I'll switch at some point
+  var defer = new promise.Promise(function(res, rej) {
+    resolve = res;
+  });
+  var stop_ids = {};
+  // this file is 30-something-goddamn megabytes, guess I'll stream this one
```

Comment:
Nice, I was trying to avoid parsing all 30MB into memory. This looks like what I want. I'll convert to use it, seeing as I've also been relying on bad things like no quotes existing in the CSVs. :tropical_drink: